nightfall-deployer/contracts/DerParser.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: CC0-1.0
+
+pragma solidity ^0.8.3;
+
+// This contract can parse a DER object, such as a suitably encoded SSL certificate
+import 'hardhat/console.sol';
+
+contract DERParser {
+    struct DecodedTlv {
+        uint256 start;
+        uint256 headerLength;
+        Tag tag;
+        uint256 length;
+        bytes value;
+        bytes octets;
+        uint256 depth;
+    }
+    struct Tag {
+        bool isConstructed;
+        bytes1 tagType;
+    }
+
+    uint256 constant MAX_DEPTH = 5;
+
+    /*
+  Parses the input tag.
+  */
+    function getTlvTag(bytes1 tagByte, uint256 pointer)
+        private
+        pure
+        returns (
+            Tag memory,
+            uint256,
+            uint256
+        )
+    {
+        bytes1 tagClass = tagByte & 0xC0;
+        bool isConstructed = (tagByte & 0x20) != 0;
+        bytes1 tagType = tagByte & 0x1F;
+        uint256 headerLength = 0;
+        require(tagType < 0x1F, 'Tag is Long Form, which is not supported');
```

Comment:
I think we have started to add the name of the contract at the beginning of error messages. You might want to apply this to all new instances.

```suggestion
        require(tagType < 0x1F, 'DerParser: Tag is Long Form, which is not supported');
```